feat: Add animation preview

Leeyah-123 opened the pull request:
# Fixes Issue

My PR closes #373 

# 👨‍💻 Changes proposed(What did you do ?)

I added an animation preview container to the animation generator and hooked it up to a function called "displayAnimationPreview", that I created in the animation.ts file. I added two new functions to getElements.ts to select the new elements I added to the HTML. I also added a bit of CSS to style the animation preview container, but I would love to hear opinions on other design options.
Lastly, added a button in the animation generator section to toggle between displaying the animation preview container.

# ✔️ Check List (Check all the applicable boxes)
<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.

##  Note to reviewers
I would love to hear your feedback on the functions I added and ways to improve them if possible, and I also would love to hear feedback on the placement of the animation preview box as Im not so satisfied with how it looks but I'm not sure where else to place it.

# 📷 Screenshots
![animation-preview-slider](https://user-images.githubusercontent.com/101122860/215324021-11621c72-ca4f-46e6-bfb9-339e74316e1e.png)

Dun-sin commented:
Hi, there's so many things wrong with this PR
 1. Build failed, please fix that
 2. You didn't install any packages, so I don't see why you committed package.json
 3. The position of the preview is very bad, see how preview is done for the shadow generators
 4. We don't need the option to hide or show preview

Leeyah-123 commented:
I have removed the show/hide preview option and all related functions.
I have also changed the display of the animation preview container.
I didn't commit the package.json file, the build keeps failing due to the absence of a package.json file.
Could you enlighten me on why we're not to commit the package.json file?

Dun-sin commented:
> I have removed the show/hide preview option and all related functions.
> I have also changed the display of the animation preview container.
> I didn't commit the package.json file, the build keeps failing due to the absence of a package.json file.
> Could you enlighten me on why we're not to commit the package.json file?

You actually have to delete this PR to fix that, i don't know any other way

Reading what's here would have saved you alot of problems right now -> https://github.com/Dun-sin/Code-Magic#-what-you-have-to-do-to-contribute

Dun-sin commented:
> > I have removed the show/hide preview option and all related functions.
> > I have also changed the display of the animation preview container.
> > I didn't commit the package.json file, the build keeps failing due to the absence of a package.json file.
> > Could you enlighten me on why we're not to commit the package.json file?
> 
> You actually have to delete this PR to fix that, i don't know any other way
> 
> Reading what's here would have saved you alot of problems right now -> https://github.com/Dun-sin/Code-Magic#-what-you-have-to-do-to-contribute
> 
> 

Following it will probably help you make the proper choices👍🏽

Dun-sin commented:
> You actually have to delete this PR to fix that, i don't know any other way
> 
> Reading what's here would have saved you alot of problems right now -> https://github.com/Dun-sin/Code-Magic#-what-you-have-to-do-to-contribute
> 
>

So following it will probably save you right now

Dun-sin commented:
Also the reason why you shouldn't commit package.json is that, you doing npm install, may install an update of the package, which could lead to problems in the long run. So except you actually installed something like a library, it's advisable to not commit package files.

As for why it deleted, it's because you committed it and then deleted it, which shows that you want to permanently delete the file from the original repo(which it already deleted in your PR, causing vercel not to see it)

To avoid this, you could have just not committed the package files in the first place. Saving you the stress
